### DISCUSSION TITLE
Problem with PWM

### DISCUSSION THREAD

**Discussion by segio-1122** [arkypita/LaserGRBL#863] [Q&A]
I have a problem with my LASER (model: ACAN A12) 15W.
Ok First of all the Laser works only in (VECTORIZE!) 

The G-CODE works correctly I observe the line by line that the code is executed as-is, it turns on M3 with the defined value and M5 turn-off the LASER, but when I put it in the "line-to-line tracing" function the LASER does not turn on.

The first thing I did was check the code and type line by line and it responds, and indeed it does, even if I wrote for example "M3 S100" and so on until "M3 S700" as an example I see that the beam of light is variable.

The next thing was to make a measurement with the multimeter and see the voltages that it handles and that I respond as programmed then 0 is 0 volts and 1000 is 5 volts, the variables that are between 0 to 1000 really when measuring are accurate to the calculations, then I realized that I had no problem, so I decided to take an "image.jpg" and send it the laser I see that the PIN "-Z" of the “CNC shield” which is the PIN 11 of my "Arduino ONE the voltage there is correct but the laser stay on OFF.

![WhatsApp Image 2020-01-27 at 10 15 11](https://user-images.githubusercontent.com/60363900/73221922-a9694580-4127-11ea-889b-e8305cbb2688.jpeg)

MY CONFIG:

![config](https://user-images.githubusercontent.com/60363900/73221945-bb4ae880-4127-11ea-9eaf-69ff8a90295f.png)

**Comment by StuartB4** [comment]
You should have M4 to turn the laser on in Target Image settings. 

![Uerernti4545tled](https://user-images.githubusercontent.com/16980026/73222519-56a98100-415b-11ea-842e-f319181be479.jpg)

**Comment by segio-1122** [comment]
Hi! StuartB4 imminently I read your post I tried but the problem continues  :(

**Comment by segio-1122** [comment]
Ok thanks arkypita

**Comment by segio-1122** [comment]
well now I tried another thing, I changed the parameter $32=1 to $32=0, and now happens something new, when I put on "line to line tracing" the LASER moves on, but only turning-on in the perimeter, but not in the fill, another thing that I observed is missing the command M4 in the final menu from the image. and when I return all again $32=0 to $32=1 and put the LASER on "line to line tracing"  the LASER moves but don´t a turn-on in nothing even on the perimeter.
